central/cluster/datastore/datastore_impl.go

```diff
@@ -312,6 +312,10 @@ func (ds *datastoreImpl) AddCluster(ctx context.Context, cluster *storage.Cluste
 	if err := checkWriteSac(ctx, cluster.GetId()); err != nil {
 		return "", err
 	}
+	_, found := ds.nameToIDCache.Get(cluster.Name)
+	if found {
+		return "", errox.AlreadyExists.Newf("cannot add cluster %s, it already exists", cluster.GetName())
```

Comment:
I think the message could be clearer because It is the cluster name that is conflicting and not the cluster config. maybe something like `cluster name must be unique. Cannot add cluster {name} because a cluster {stored cluster name} ({id}) already exists`

Reply:
Updated the message.